src/server/main.py

```diff
@@ -136,15 +119,11 @@ def register() -> Response:
     if not email or not password:
         return make_response({"message": "Invalid Json format"}, 202)
 
-    try:
-        connection: Connection = getDatabaseConnection()
-        sql: str = "SELECT * FROM users WHERE email=?"
-        cursor: Cursor = connection.cursor()
-        cursor.execute(sql, (email,))
-        user_information: list[tuple] = cursor.fetchall()
-        connection.close()
-    except SqlError as e:
-        return make_response({"message": f"Database connection error: {e}"}, 501)
+    with DatabaseProvider.query("SELECT * FROM users WHERE email=?", (email,)) as response:
+        if response.message is not Message.OK:
+            return make_response({"message": f"Internal error: {response.message}"}, 501)
+
+        user_information: list[tuple[str, str]] = response.data
```

Comment:
I think it should be just
```suggestion
        user_information: list[tuple] = response.data
```
as we don't care about the fields this query returns